python/ml4ir/base/pipeline.py

```diff
@@ -474,6 +472,26 @@ def finish(self, job_status, job_info):
             self.file_io.copy_to_hdfs(self.models_dir_local, self.models_dir, overwrite=True)
             self.file_io.copy_to_hdfs(self.logs_dir_local, self.logs_dir, overwrite=True)
 
+        if self.args.write_tf_model_dir:
```

Comment:
Could be a good idea to move all that to a separate method (rules of thumb for me: a function should do a single thing).